spec/lang/syntax.md

```diff
@@ -201,6 +201,7 @@ pub enum BinOp {
     Rel(RelOp),
 
     /// Add a byte-offset to a pointer (with or without inbounds requirement).
+    /// For non-Sized pointees this leaves the metadata untouched.
```

Comment:
This should require a sized pointee.

Reply:
Oh I guess I was looking at the new [byte_offset](https://doc.rust-lang.org/std/primitive.pointer.html#method.byte_offset-1) method, where it explicitly reassigns the metadata.

Reply:
Yeah but that's happening in Rust; the underlying MIR / MiniRust operation requires a sized pointee. If you look at the implementation of that function you can see that it casts the pointer to `u8`.